### DISCUSSION TITLE
Joining SFC and leadership structure

### DISCUSSION THREAD

**Discussion by ethomson** [libgit2/libgit2#5878] [General]
The libgit2 project has applied to join the [Software Freedom Conservancy](https://sfconservancy.org) as a member project, and we have been offered membership in the SFC.  The Software Freedom Conservancy exists to provide a (non-profit) home and infrastructure for free and open-source projects, and will help us take care of the non-code issues that the project has.  For example: they can act in a financial or legal capacity for us.  Importantly, the [git project](https://git-scm.com/sfc) is part of the Software Freedom Conservancy already, so this is a natural continuation of our relationship with both the git project and the SFC.

We will be negotiating an agreement similar to this example: https://sfconservancy.org/projects/apply/ConservancyFSATemplate.pdf when we join.

Importantly, we need to create a project leadership structure.  Throughout the course of the libgit2 project we have not had a formal leadership structure documented, though in the last few years we have had a somewhat assumed leadership committee of [Carlos Martín-Nieto](https://github.com/carlosmn/) and [Patrick Steinhardt](https://github.com/pks-t/), and [me](https://github.com/ethomson/).

Joining the Software Freedom Conservancy as a member project requires us to clearly document a leadership structure.  This leadership committee will oversee the _non-technical_ aspects of libgit2.  This is not a committee that decides when - or whether - pull requests get merged: instead it will handle the financial and legal aspects of the project.  Ensuring that the (thankfully few) bills get paid, etc.

To give you an idea of the things that the git project worries about, [peff](https://github.com/peff) has posted status updates for [2017](https://public-inbox.org/git/20170202024501.57hrw4657tsqerqq@sigill.intra.peff.net/) and [2018](https://public-inbox.org/git/20180306231609.GA1632@sigill.intra.peff.net/).

We are proposing that libgit2 adopt a project leadership committee that consists of:

* 5 people who have an interest in the project
* People who are contributors to the project (either writing code, documentation, testing, etc) or are users of the project.  We want to encourage people who use and rely on libgit2 to take a leadership role and ensure that the interests of end-users are represented.
* No more than two people from the same company can be on the board.  We want to both ensure that libgit2 is an independent project and not controlled by a single company, while ensuring that core contributors have an opportunity to move between employers.  (As one would expect: many of the contributors have worked at a small number of companies.  Concretely, Carlos and Patrick have both worked at the same company, Carlos and I have worked together at two separate companies.)

But having only three people in the de facto leadership committee means that we need to grow our committee.  I propose that we take nominations: if you are a user or contributor and you think that you have seen leadership from someone and you think that you want them to represent the project, then please nominate them.  If you would like to help manage the libgit2 project, please nominate yourself.

**Comment by pks-t** [comment]
I'd like to propose @tiennou as another member of the PLC. He has been very active recently and produced a lot of good work, and thus is a valuable member of the project.

**Comment by ethomson** [comment]
Thanks, @pks-t.  I've also been incredibly impressed by the work that @tiennou has shown, and I appreciate both his judgement and technical leadership.

**Comment by tiennou** [comment]
First, thanks for the vote of confidence ! I'm currently deep into work for a few days (onlining brand new websites is stressful), so I wanted to take a few days to think it through. I'm slightly uncertain of the commitment required of a board member so I'm slightly cautious, but on the other hand if it allows us to join the SFC and help grow the project I'm 👍.

**Comment by ethomson** [comment]
That's a great question, @tiennou.  This is new territory for all of us, so it's a little hard to know the exact time investments required, but I would like to make sure that it is _not_ burdensome.  Thankfully, we don't have a ton of legal issues, or financial issues, so I don't expect this to be a big problem.

We can ask the SFC folks, or another project (like git) to weigh in on how much process it added, but I definitely don't get the feeling that it is onerous at all.

**Comment by ethomson** [comment]
I've been giving this some thought, and I'd like to nominate Tyler Ang-Wanek (@implausible). 

 As he was regrettably not able to travel to spain for the last [libgit2 summit](https://github.com/libgit2/discussions/issues/3), some of you may not have met him

Tyler is one of the core contributors to [NodeGit](https://github.com/nodegit/nodegit), which is of course the libgit2 binding for node.js.  He also works on a project ([GitKraken](https://gitkraken.com/), which uses both NodeGit and libgit2.  And he's contributed to libgit2, working to [add multi-threaded checkout](https://github.com/libgit2/libgit2/pull/4205) (which I, regrettably, need to take the time to review.  Sorry, Tyler.)

I've talked with Tyler quite a bit, both when he was working on the parallelization pull request, and about how libgit2 works with NodeGit.  I've found him to be thoughtful about both technical and non-technical aspects of the projects.

**Comment by implausible** [comment]
Following @tiennou, I really appreciate the confidence. Thanks for the consideration.

**Comment by rhuijben** [comment]
Any news on the schedule/progress?

More help needed?

**Comment by ethomson** [comment]
Hi @rhuijben - sorry that things have been moving slowly, we've been trying to work around travel schedules.  I'll post an update early next week.

**Comment by rhuijben** [comment]
We are roughly 2 months further... Any update?
